sap/note/ini_sections.go

```diff
@@ -94,41 +94,67 @@ var isNrreq = regexp.MustCompile(`^NRREQ_\w+$`)
 
 // GetBlkVal initialise the block device structure with the current
 // system settings
-func GetBlkVal(key string, cur *param.BlockDeviceQueue) (string, error) {
+func GetBlkVal(key string, cur *param.BlockDeviceQueue) (string, string, error) {
 	newQueue := make(map[string]string)
 	newReq := make(map[string]int)
 	retVal := ""
+	info := ""
 
 	switch {
 	case isSched.MatchString(key):
 		newIOQ, err := cur.BlockDeviceSchedulers.Inspect()
 		if err != nil {
-			return "", err
+			return "", info, err
 		}
 		newQueue = newIOQ.(param.BlockDeviceSchedulers).SchedulerChoice
 		retVal = newQueue[strings.TrimPrefix(key, "IO_SCHEDULER_")]
 		cur.BlockDeviceSchedulers = newIOQ.(param.BlockDeviceSchedulers)
 	case isNrreq.MatchString(key):
 		newNrR, err := cur.BlockDeviceNrRequests.Inspect()
 		if err != nil {
-			return "", err
+			return "", info, err
 		}
 		newReq = newNrR.(param.BlockDeviceNrRequests).NrRequests
 		retVal = strconv.Itoa(newReq[strings.TrimPrefix(key, "NRREQ_")])
 		cur.BlockDeviceNrRequests = newNrR.(param.BlockDeviceNrRequests)
 	}
-	return retVal, nil
+	return retVal, info, nil
 }
 
 // OptBlkVal optimises the block device structure with the settings
 // from the configuration file
-func OptBlkVal(key, cfgval string, cur *param.BlockDeviceQueue) string {
+func OptBlkVal(key, cfgval string, cur *param.BlockDeviceQueue, bOK map[string][]string) (string, string) {
+	info := ""
+	if cfgval == "" {
+		return cfgval, info
+	}
 	sval := cfgval
 	switch {
 	case isSched.MatchString(key):
-		sval = strings.ToLower(cfgval)
-		opt, _ := cur.BlockDeviceSchedulers.Optimise(sval)
-		cur.BlockDeviceSchedulers = opt.(param.BlockDeviceSchedulers)
+		oval := ""
+		sfound := false
+		dname := regexp.MustCompile(`^IO_SCHEDULER_(\w+)$`)
+		bdev := dname.FindStringSubmatch(key)
+		for _, sched := range strings.Split(cfgval, ",") {
+			sval = strings.ToLower(strings.TrimSpace(sched))
+			if !param.IsValidScheduler(bdev[1], sval) {
+				//system.WarningLog("'%s' is not a valid scheduler for device '%s', skipping.", sval, bdev[1])
```

Comment:
@angelabriel I recommend removing commented code. If it is not needed, simply remove it to keep it clean, or include a proper comment here.

Reply:
yes, your right. This is a leftover from the discussion with Sören and now useless. I will remove it. Thanks.